R/install-gitlab.R

```diff
@@ -71,7 +71,7 @@ remote_download.gitlab_remote <- function(x, quiet = FALSE) {
   dest <- tempfile(fileext = paste0(".tar.gz"))
 
   src_root <- build_url(x$host, "api", "v4", "projects", utils::URLencode(paste0(x$username, "/", x$repo), reserved = TRUE))
-  src <- paste0(src_root, "/repository/archive.tar.gz?ref=", utils::URLencode(x$ref, reserved = TRUE))
```

Comment:
I think there was a misunderstanding. IMO you only need to change the `?ref` to `?sha` and leave everything else as is (or rather revert all other changes you made in this commit).